sidh/api.go

```diff
@@ -0,0 +1,184 @@
+package sidh
+
+import (
+	"errors"
+	. "github.com/cloudflare/p751sidh/p751toolbox"
+	"io"
+)
+
+// I keep it bool in order to be able to apply logical NOT
+type KeyVariant uint
+type PrimeFieldId uint
+
+// Id's correspond to bitlength of the prime field characteristic
+// Currently PF_P751 is the only one supported by this implementation
+const (
+	PF_P503 PrimeFieldId = iota
```

Comment:
Would be better to call them FP_503, FP_xxx, etc. to stay consistent with Fp751Element structure name